upgrade/satellite.py

```diff
@@ -74,189 +71,150 @@ def satellite6_setup(os_version):
-        if to_version == '6.1':
-            run('katello-installer --upgrade')
-        else:
-            run('satellite-installer --scenario satellite --upgrade')
-        postup_time = datetime.now().replace(microsecond=0)
-        logger.highlight('Time taken for Satellite Upgrade - {}'.format(
-            str(postup_time-preup_time)))
+            repository_setup("[sat6]",
+                             "satellite 6",
+                             base_url, 1, 0)
+        upgrade_task()
     # Rebooting the satellite for kernel update if any
     reboot(180)
     host_ssh_availability_check(env.get('satellite_host'))
     # Test the Upgrade is successful
-    run('hammer ping', warn_only=True)
-    run('katello-service status', warn_only=True)
+    upgrade_validation()
 
 
-def satellite6_zstream_upgrade():
-    """Upgrades Satellite Server to its latest zStream version
+def foreman_maintain_upgrade(base_url):
+    """
+    The purpose of this function is to setup the foreman-maintain and perform the
+    foreman-mantain upgrade"
+    :param base_url: It is used to check the repository selection whether
+    it from CDN or from Downstream
+    """
+    if base_url is None:
+        os.environ['DISTRIBUTION'] = "CDN"
+    else:
+        os.environ['DISTRIBUTION'] = "DOWNSTREAM"
+    # setup foreman-maintain
+    setup_foreman_maintain()
+    preup_time = datetime.now().replace(microsecond=0)
+    # perform upgrade using foreman-maintain
+    upgrade_using_foreman_maintain()
+    postup_time = datetime.now().replace(microsecond=0)
+    logger.highlight('Time taken for Satellite Upgrade - {}'.format(
+        str(postup_time - preup_time)))
 
-    Note: For zstream upgrade both 'To' and 'From' version should be same
 
-    FROM_VERSION
-        Current satellite version which will be upgraded to latest version
-    TO_VERSION
-        Next satellite version to which satellite will be upgraded
-    PERFORM_FOREMAN_MAINTAIN_UPGRADE
-        use foreman-maintain for satellite upgrade
+def enable_disable_repo(enable_repo_name, disable_repo_name):
+    """
+    The purpose of this function is to enable and disable the
+    repository as per requirements.
+    :param enable_repo_name: This will take the list of
+    repository which you are going to enable
+    :param disable_repo_name: This will take the
+    list of repository which you are going to disable
+    """
+    if disable_repo_name:
+        [disable_repos('{}'.format(repo), silent=True) for repo in disable_repo_name]
+    if enable_repo_name:
+        [enable_repos('{}'.format(repo)) for repo in disable_repo_name]
+
+
+def repository_cleanup(repo_name):
+    """
+    The purpose of this function to perform the repository cleanup on the basis of
+    their name.
+    :param repo_name: repository name
+    """
+    for fname in os.listdir('/etc/yum.repos.d/'):
+        if repo_name in fname.lower():
+            os.remove('/etc/yum.repos.d/{}'.format(fname))
+
+
+def upgrade_task():
```

Comment:
better naming is required something like manual_upgrade or nonfm_upgrade

Reply:
Agree, I am keeping nonfm_upgrade.